libs/sdk-ui-dashboard/src/plugins/plugin.ts

```diff
@@ -134,88 +244,44 @@ export interface IDashboardPlugin extends IDashboardPluginMetadata {
-     * @param propsSoFar - contains customization props constructed so far during the dashboard initialization process; the
-     *  function MUST NOT modify these props. The props are provided here in case the plugin wants to introspect existing
-     *  customization and alter its own behavior based on it.
+     * @param customize - API through which you can register dashboard customizations
+     * @param handlers - an object containing event handler registration functions
      *
      * @returns this plugin's contribution to the dashboard customization props; if undefined, there is no contribution
      */
-    createCustomizationProps?(
+    register?(
         pluginCtx: IDashboardPluginContext,
-        propsSoFar: Readonly<IDashboardCustomizationProps>,
-    ): IDashboardCustomizationProps | undefined;
+        customize: IDashboardCustomizer,
+        handlers: IDashboardEventHandlers,
+    ): void;
 
     /**
-     * This function will be called before the dashboard rendering starts. The plugin can provide its instance
-     * of {@link IDashboardThemingProps} that can be used to influence the theme applied for this dashboard.
-     *
-     * The plugin MAY introspect the theme constructed so far and then return its own customization props. The
-     * merging of the theme is not the responsibility of the plugin or this function.
-     *
-     * @param pluginCtx - context in which this plugin operates
-     * @param themeSoFar - contains theme constructed so far during the dashboard initialization process; the
-     *  function MUST NOT modify these props. The props are provided here in case the plugin wants to introspect existing
-     *  customization and alter its own behavior based on it.
-     */
-    createThemingProps?(
-        pluginCtx: IDashboardPluginContext,
-        themeSoFar: Readonly<IDashboardThemingProps>,
-    ): IDashboardThemingProps | undefined;
-
-    /**
-     * This function will be called to obtain callbacks that the plugin wants to register with the dashboard.
-     *
-     * @param pluginCtx - context in which this plugin operates
-     */
-    createEventingProps?(pluginCtx: IDashboardPluginContext): IDashboardEventing | undefined;
-
-    /**
-     * This function will be called right before the dashboard rendering starts. The plugin can provide a React
-     * component that will be added as child of the Dashboard itself. The dashboard will render this custom
-     * component right above the dashboard layout itself and below the filter bar.
+     * This function will be called when user navigates away from the dashboard that uses an instance of
+     * this plugin. At this point, the plugin SHOULD perform any essential cleanup.
      *
      * @param pluginCtx - context in which this plugin operates
      */
-    getDashboardChildComponent?(pluginCtx: IDashboardPluginContext): React.ComponentType;
+    onPluginUnload?(pluginCtx: IDashboardPluginContext): void;
```

Comment:
I wonder if this should be async. What if the cleanup includes some network request for example. Do we want to wait for it, or is "fire and forget" ok here?

Reply:
good point